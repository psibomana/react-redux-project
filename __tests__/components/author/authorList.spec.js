import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import AuthorList from '../../../src/components/author/authorList'

describe('Author List Test', () => {
  let wrapper, authors;
  beforeEach(() => {
    authors = [
      {
        id: 'cory-house',
        firstName: 'Cory',
        lastName: 'House'
      },
      {
        id: 'scott-allen',
        firstName: 'Scott',
        lastName: 'Allen'
      },
      {
        id: 'dan-wahlin',
        firstName: 'Dan',
        lastName: 'Wahlin'
      }
    ]

    wrapper = shallow(
      <AuthorList authors={authors}/>
    )
  })
  it('should render as expected', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  })
})
