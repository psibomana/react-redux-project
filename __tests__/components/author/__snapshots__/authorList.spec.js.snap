// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author List Test should render as expected 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th>
        ID
      </th>
      <th>
        First Name
      </th>
      <th>
        Last Name
      </th>
    </tr>
  </thead>
  <tbody>
    <AuthorListRow
      author={
        Object {
          "firstName": "Cory",
          "id": "cory-house",
          "lastName": "House",
        }
      }
      key="cory-house"
    />
    <AuthorListRow
      author={
        Object {
          "firstName": "Scott",
          "id": "scott-allen",
          "lastName": "Allen",
        }
      }
      key="scott-allen"
    />
    <AuthorListRow
      author={
        Object {
          "firstName": "Dan",
          "id": "dan-wahlin",
          "lastName": "Wahlin",
        }
      }
      key="dan-wahlin"
    />
  </tbody>
</table>
`;
